File: senz-client-samples/ruby/receiver.rb
require_relative 'utils/AesUtils'
require_relative 'utils/imageUtils'
require_relative 'client'

sharedKey  = "LGWlhb329Y09oluI"
imagePathToReceive = "received.jpg"

#registering the device
puts "registering the device.."
timeS = timeStamp()
regmsg = "SHARE #pubkey KEY @senz #time %s ^dev2 signature" % [timeS]
sendMessage(regmsg)

#receiving a Message
puts "receiving a message"
msg = receiveMessage
arr = msg.split(" ")
byteString = arr[2..-8]
byteString = byteString.join(" ")
puts "image received"
puts byteString


#decoding a message
aes = AesUtil.new(sharedKey)
decryptedByteString = aes.decode(byteString)
#puts decryptedByteString

#saving image
stringToImage(decryptedByteString, imagePathToReceive)
puts "check your directory"

File: senz-client-samples/java/Receiver.java
public class Receiver {
  static String deviceName = "dev2";
  static String sharedKey = "LGWlhb329Y09oluI";
  static String imagePathToStore = "received.jpg";

  public static void main(String args[]) {
    Client client = new Client();
    Image_Utils imageUtils = new Image_Utils();
    AES_Utils aesUtils;

    // Register device
    System.out.println("Registering...");
    String msg = "SHARE #pubkey key @senz #time "+client.getTimeStamp()+" ^"+deviceName+" signature";
    client.sendMessage(msg);

    // Receive message
    System.out.println("Ready to receive image");
    String reply = client.receiveMessage();
    String byteString = reply.split(" ")[2];
    System.out.println("Image received...");

    // Decode using aes crypto
    aesUtils = new AES_Utils(sharedKey);
    byteString = aesUtils.decrypt(byteString);

    // Save Image
    imageUtils.stringToImage(byteString, imagePathToStore);
    System.out.printf("Image saved (%s)\n", imagePathToStore);

    // Unregister device
    System.out.println("Unregistering...");
    msg = "UNSHARE #pubkey key @senz #time "+client.getTimeStamp()+" ^"+deviceName+" signature";
    client.sendMessage(msg);
  }
}
